### PR TITLE
feat(admin-tool): merge CSV toolbox into Give core #2875

### LINKED ISSUE #2875: feat(admin-tool): merge CSV toolbox into Give core
## Issue Overview
The CSV toolbox has proven useful enough to be merged into core: https://github.com/WordImpress/Give-CSV-Toolbox

## Task List
- [x] Remove the old Give Donation History functionality 
- [x] update code to support with Give 2.0 DB upgrade 
- [x] Add export Donation Link to submenu pages that export Donation History 
- [x] Update Give-CSV-Toolbox functionality so admin can select all form option and then export the donation ( Hide FFM, Custom Filed and Hidden Custom Field Columns when all Donation form is selected from DropDown  )
- [x] Add export Donation Link to submenu pages that export Donation History 
- [x] Let's also make sure that FFM, Custom Filed, and Hidden Custom Field Columns sections only display if there is actually a checkbox available within the section
- [x] Fix issues https://github.com/WordImpress/Give/issues/2978 ( Issues closed not able to reproduce the issues again )
- [ ] Delete Give-CSV-Toolbox repository and write a blog post about it 

## Related Issues

- [x] https://github.com/WordImpress/Give-CSV-Toolbox/issues/7
- [ ] https://github.com/WordImpress/Give/issues/2978

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #2875 

## How Has This Been Tested?
Manually Tested after removing the Export Donation History Code and then add the CSV ToolBox Code to Give Core 2.1 and exporting the donation from the CSV
## Task List
- [x] Remove the old Give Donation History functionality 
- [x] update code to support with Give 2.0 DB upgrade 
- [x] Add export Donation Link to submenu pages that export Donation History 
- [x] Update Give-CSV-Toolbox functionality so admin can select all form option and then export the donation ( Hide FFM, Custom Filed and Hidden Custom Field Columns when all Donation form is selected from DropDown  )
- [x] Add export Donation Link to submenu pages that export Donation History 
- [x] Let's also make sure that FFM, Custom Filed, and Hidden Custom Field Columns sections only display if there is actually a checkbox available within the section
- [x] Fix issues https://github.com/WordImpress/Give/issues/2978 ( Issues closed not able to reproduce the issues again )
- [ ] Delete Give-CSV-Toolbox repository and write a blog post about it
 
## Screenshots (jpeg or gifs if applicable):
View the Export Page
image

With out selecting the Donation Form:
image

When selecting the Donation Form:
image

## Types of changes
<!-- New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@raftaar1191 Arrange checkboxes in columns form like below image:

image

File: assets/src/js/admin/admin-settings.js [hunk 1]
```diff
@@ -150,4 +150,132 @@ jQuery(document).ready(function ($) {
...
+	$( '.give-export_donations #give-export_donations-form #give_form_for_csv_export' ).chosen().change( function () {
+
+		$( '.give-export-donations-hide' ).addClass( 'give-hidden' );
+
+		jQuery(document).trigger('give_export_donations_form_change' );
+
+		var give_form_id;
+
+		// Check for form ID.
+		if ( ! (
```
Comment (ravinderk) [inline assets/src/js/admin/admin-settings.js]: @raftaar1191 improve code formatting here.

File: assets/src/js/admin/admin-settings.js [hunk 2]
```diff
@@ -150,4 +150,132 @@ jQuery(document).ready(function ($) {
...
+				form_id: give_form_id,
+				action: 'give_export_donations_get_custom_fields'
+			},
+			success: function ( response ) {
+
+				if ( response ) {
+					output_give_donations_fields( response );
+
+				} else {
+					alert( 'An AJAX error occurred.' );
```
Comment (ravinderk) [inline assets/src/js/admin/admin-settings.js]: @raftaar1191 this must be translatable

**Comment by raftaar1191** [comment]
@ravinderk updated the PR as you have mentioned in **requested changes**


image

File: assets/src/js/admin/admin-settings.js [hunk 3]
```diff
@@ -150,4 +150,127 @@ jQuery(document).ready(function ($) {
 			}
 		})
 	});
+
+	/**
+	 * Ajax call to get donation fields.
+	 */
+	$( '.give-export_donations #give-export_donations-form #give_form_for_csv_export' ).chosen().change( function () {
```
Comment (DevinWalker) [inline assets/src/js/admin/admin-settings.js]: @raftaar1191 please remove this from create a new JS file called `admin-exports.js`

Comment (raftaar1191) [reply assets/src/js/admin/admin-settings.js]: Okay @DevinWalker doing that

**Comment by DevinWalker** [comment]
## Change Requests

### 1. Vertically align to the top top align rows



### 2. Fix indent



### 3. Make this export the first option

**Comment by raftaar1191** [comment]
@DevinWalker  Update PR with all the suggestion that you have made

**Comment by DevinWalker** [comment]
@raftaar1191 can you fix the conflict in this PR?

**Comment by ravinderk** [comment]
@DevinWalker merge coflict resolved.

**Pull request merged**